OpenRA.Platforms.Default/Sdl2PlatformWindow.cs

```diff
@@ -123,8 +125,31 @@ public Sdl2PlatformWindow(Size requestEffectiveWindowSize, WindowMode windowMode
 				else if (Platform.CurrentPlatform != PlatformType.OSX)
 				{
 					float scale = 1;
-					var scaleVariable = Environment.GetEnvironmentVariable("OPENRA_DISPLAY_SCALE");
-					if (scaleVariable != null && float.TryParse(scaleVariable, out scale))
+
+					// Launch the game with OPENRA_DISPLAY_SCALE to force a specific scaling factor
+					// Otherwise fall back to GDK_SCALE or parsing the x11 DPI configuration
+					var scaleVariable = Environment.GetEnvironmentVariable("OPENRA_DISPLAY_SCALE") ?? Environment.GetEnvironmentVariable("GDK_SCALE");
+					if (scaleVariable == null || !float.TryParse(scaleVariable, NumberStyles.Float, NumberFormatInfo.InvariantInfo, out scale))
+					{
+						// Attempt to automatically detect DPI
+						try
+						{
+							var psi = new ProcessStartInfo("/usr/bin/xrdb", "-query");
+							psi.UseShellExecute = false;
+							psi.RedirectStandardOutput = true;
+							var p = Process.Start(psi);
+							var lines = p.StandardOutput.ReadToEnd().Split('\n');
+
+							int dpi;
+							foreach (var line in lines)
+								if (line.StartsWith("Xft.dpi"))
+									if (int.TryParse(line.Substring(8), out dpi))
```

Comment:
Nothing major, but this could simplify to `if (line.StartsWith("Xft.dpi") && int.TryParse(line.Substring(8), out dpi))`.